05_syntax_lab/03.py

```diff
@@ -1,7 +1,12 @@
-"""
-Write a program that randomizes a number
-and prints the sum total of its digits.
-For example if the number was: 2345
-The result should be: 14
-"""
-
+################### Part 3
+L = 10000
+r = randint(1, L)
+str_r = str(r)
+print('The string number is {}'.format(str_r) )
+num_digits = len(str_r)
+print('The number %d contains %d digits' % (r,num_digits) )
+sum = 0
+for i in range(0, num_digits):
+    sum += int(str_r[i])
+
+print('The digits sum is {}'.format(sum) )

```

Comment:
* You don't need an index to iterate over characters of a string you can write ```for c in str_r```
* It would be better to consistently use ```.format``` instead of the old ```%``` syntax.